flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorEvent.java

```diff
@@ -24,4 +24,12 @@
  * Root interface for all events sent between {@link OperatorCoordinator} and an {@link
  * OperatorEventHandler}.
  */
-public interface OperatorEvent extends Serializable {}
+public interface OperatorEvent extends Serializable {
+    /**
+     * @return true if event is optional and an occasional loss or inability to deliver that event
+     *     doesn't affect the job's correctness.
+     */
+    default boolean isOptional() {
+        return false;
```

Comment:
NIT: 
How about returning an `enum DeliveryRequirement { MANDATORY, OPTIONAL }`
or `void handleDeliveryFailure(Exception failure);`?

For more flexibility (e.g. tolerating up to N failures)